frontend/src/hooks/subscriptions.ts

```diff
@@ -1,22 +1,66 @@
 import * as React from "react";
+import * as ReactQuery from "react-query";
 
 import { SubscriptionTopicType, SubscriptionTopicTypes } from "../types/common";
 import { subscribe } from "../libraries/transport";
 import { useNetworkContext } from "./use-network-context";
 
+type UseSubscriptionResult<Topic extends SubscriptionTopicType> = Omit<
```

Comment:
We may change the resulting type eventually, but this one seems like a good start